src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,17 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
+        if (user == null) {
+            throw new InvalidDataException("invalid data");
+        } else if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new InvalidDataException("problem with login");
+        } else if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new InvalidDataException("problem with password");
+        } else if (user.getAge() < 18) {
+            throw new InvalidDataException("problem with age");
+        } else if (storageDao.get(user.getLogin()) == null) {
+            return storageDao.add(user);
```

Comment:
The condition `storageDao.get(user.getLogin()) == null` checks if the user does not exist in the storage. If the user does not exist, it should proceed to add the user, which is correct. However, the method should return the newly added user instead of returning `null` at the end. Consider changing the logic to return the user after successful registration or handle the case where the user already exists.